### PR TITLE
ENH: update `dpnp.fft`; MKL integration

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
* Integrated mkl fft routines for `complex-to-complex` (`complex128`, `complex64` i/o dtypes) cases for 1,2,3 dim arrays.
* extended test cases for `dpnp.fft.fft`.

## TODO
* `real-to-complex` cases (on other PR(s))
* update mkl using for 2 and 3 array dims. (MKL FFT supports only 1,2,3 dim arrays [fft2, ifft2 and etc.]). (on other PR(s))
* Fix dpnp own fft function/kernel (`dpnp_fft_fft_sycl_c`), sigsev on GPU. (on other PR(s))

## Tests
* for __CPU__: `SYCL_DEVICE_FILTER=cpu pytest -v tests/test_fft.py::test_fft`
```bash
tests/test_fft.py::test_fft[complex128] PASSED
tests/test_fft.py::test_fft[complex64] PASSED
tests/test_fft.py::test_fft[float32] PASSED
tests/test_fft.py::test_fft[float64] PASSED
tests/test_fft.py::test_fft[int32] PASSED
tests/test_fft.py::test_fft[int64] PASSED
```
* for __GPU__: `SYCL_DEVICE_FILTER=gpu pytest -v tests/test_fft.py::test_fft`
```bash
tests/test_fft.py::test_fft[complex128] PASSED
tests/test_fft.py::test_fft[complex64] PASSED
tests/test_fft.py::test_fft[float32] SKIPPED (Skipping test.)
tests/test_fft.py::test_fft[float64] SKIPPED (Skipping test.)
tests/test_fft.py::test_fft[int32] SKIPPED (Skipping test.)
tests/test_fft.py::test_fft[int64] SKIPPED (Skipping test.)
```
*  cupy

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_fft.cpp [hunk 1]
```diff
@@ -42,22 +42,18 @@ template <typename _KernelNameSpecialization1, typename _KernelNameSpecializatio
...
 template <typename _DataType_input, typename _DataType_output>
-void dpnp_fft_fft_c(const void* array1_in,
-                    void* result1,
-                    const long* input_shape,
-                    const long* output_shape,
-                    size_t shape_size,
-                    long axis,
-                    long input_boundarie,
-                    size_t inverse)
+void dpnp_fft_fft_kernel_func_c(const void* array1_in,
```
Comment (shssf) [inline dpnp/backend/kernels/dpnp_krnl_fft.cpp:45]:
```suggestion
void dpnp_fft_fft_sycl_c(const void* array1_in,
```

File: dpnp/backend/kernels/dpnp_krnl_fft.cpp [hunk 2]
```diff
@@ -157,6 +143,47 @@ void dpnp_fft_fft_c(const void* array1_in,
...
+        _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+        oneapi::mkl::dft::descriptor<mkl_dft::precision::DOUBLE, mkl_dft::domain::COMPLEX> desc(result_size);
+        desc.set_value(mkl_dft::config_param::BACKWARD_SCALE, (1.0 / result_size));
+        desc.set_value(mkl_dft::config_param::PLACEMENT, DFTI_NOT_INPLACE); // enum value from math library C interface
+        // instead of mkl_dft::config_value::NOT_INPLACE
+        desc.commit(DPNP_QUEUE);
+
+        event = mkl_dft::compute_forward(desc, array_1, result);
+        event.wait();
```
Comment (shssf) [inline dpnp/backend/kernels/dpnp_krnl_fft.cpp:176]: Please move this function into `dpnp_fft_fft_mathlib_c`

Comment (samir-nasibli) [reply dpnp/backend/kernels/dpnp_krnl_fft.cpp:183]: PR in progress.

**Comment by samir-nasibli** [comment]
@Alexander-Makaryev changes are ready for review/merge. All related info provided on PR description.

**Review by Alexander-Makaryev — APPROVED** [review]
LGTM

**Pull request merged**